datalake/src/main/java/com/sequenceiq/datalake/flow/delete/SdxDeleteActions.java

```diff
@@ -189,10 +195,15 @@ protected void doExecute(SdxContext context, RdsDeletionSuccessEvent payload, Ma
                     metricService.incrementMetricCounter(MetricType.SDX_DELETION_FINISHED, sdxCluster);
                 }
                 eventSenderService.notifyEvent(context, ResourceEvent.SDX_CLUSTER_DELETION_FINISHED);
+                Optional<FlowLogWithoutPayload> lastFlowLog = flowLogService.getLastFlowLog(context.getFlowParameters().getFlowId());
                 if (flowChainLogService.isFlowTriggeredByFlowChain(
                         DatalakeResizeFlowEventChainFactory.class.getSimpleName(),
-                        flowLogService.getLastFlowLog(context.getFlowParameters().getFlowId()))) {
+                        lastFlowLog)) {
                     eventSenderService.notifyEvent(context, ResourceEvent.DATALAKE_RESIZE_COMPLETE);
+                } else if (!flowChainLogService.isFlowTriggeredByFlowChain(DatalakeResizeRecoveryFlowEventChainFactory.class.getSimpleName(), lastFlowLog)){
+                    structuredEventCleanUpService.cleanUp(payload.getResourceId());
+                } else {
```

Comment:
Doesn't this else make the above else if redundant? Effectively whether it is or isn't resize recovery the cleanup will always be called. I think we should only have the else if right?